ivy_tests/test_ivy/test_functional/test_core/test_gradients.py

```diff
@@ -202,40 +234,155 @@ def test_execute_with_gradients(
+    stop_gradients
+):
+    dtype, w = dtype_and_w
+    w = np.asarray(w,dtype=dtype)
+    effective_grad = np.asarray(effective_grad,dtype=dtype)
+    helpers.test_function(
+        dtype,
+        as_variable,
+        False,
+        native_array,
+        fw,
+        num_positional_args,
+        container,
+        instance_method,
+        "optimizer_update",
+        w=w,
+        effective_grad=effective_grad,
+        lr=lr,
+        inplace=inplace,
+        stop_gradients=stop_gradients,
+    )
+    
 # gradient_descent_update
-@pytest.mark.parametrize(
-    "ws_n_grads_n_lr_n_wsnew",
-    [(Container({"w": [3.0]}), Container({"w": [6.0]}), 0.1, Container({"w": [2.4]}))],
+@given(
+    dtype_and_w=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    as_variable=st.booleans(),
+    native_array=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="gradient_descent_update"),
+    container=helpers.list_of_length(st.booleans(), 2),
+    instance_method=st.booleans(),
+    dcdw=st.floats(),
+    lr=st.floats(min_value=0.0,max_value=1.0,allow_nan=False),
+    inplace=st.booleans(),
+    stop_gradients=st.booleans()
 )
-@pytest.mark.parametrize("dtype", ["float32"])
-@pytest.mark.parametrize("tensor_fn", [ivy.array, helpers.var_fn])
 def test_gradient_descent_update(
-    ws_n_grads_n_lr_n_wsnew, dtype, tensor_fn, device, call
+    dtype_and_w,
+    as_variable,
+    native_array,
+    num_positional_args,
+    container,
+    instance_method,
+    fw,
+    dcdw,
+    lr,
+    inplace,
+    stop_gradients
 ):
-    # smoke test
-    ws_raw, dcdws_raw, lr, ws_raw_new = ws_n_grads_n_lr_n_wsnew
-    ws = ws_raw.map(lambda x, _: ivy.variable(ivy.array(x)))
-    dcdws = dcdws_raw.map(lambda x, _: ivy.array(x))
-    ws_true_new = ws_raw_new.map(lambda x, _: ivy.variable(ivy.array(x)))
-    ws_new = ivy.gradient_descent_update(ws, dcdws, lr)
-    # type test
-    assert isinstance(ws_new, dict)
-    # cardinality test
-    for (w_new, w_true_new) in zip(ws_new.values(), ws_true_new.values()):
-        assert w_new.shape == w_true_new.shape
-    # value test
-    for (w_new, w_true_new) in zip(ws_new.values(), ws_true_new.values()):
-        assert np.allclose(ivy.to_numpy(w_new), ivy.to_numpy(w_true_new))
-    # compilation test
-    if call in [helpers.torch_call]:
-        # pytorch scripting does not support internal function definitions
-        return
-
+    dtype, w = dtype_and_w
+    w = np.asarray(w,dtype=dtype)
+    dcdw = np.asarray(dcdw,dtype=dtype)
+    helpers.test_function(
+        dtype,
+        as_variable,
+        False,
+        native_array,
+        fw,
+        num_positional_args,
+        container,
+        instance_method,
+        "gradient_descent_update",
+        w=w,
+        dcdw=dcdw,
+        lr=lr,
+        inplace=inplace,
+        stop_gradients=stop_gradients,
+    )
 
 # layerwise_gradient_descent_update
-@pytest.mark.parametrize(
-    "ws_n_grads_n_lr_n_wsnew",
+@given(
```

Comment:
When using use hard-coded ground truth values (which is necessary for compositional functions), then we should use a unique name for the test, such as `test_<func_name>_ground_truth`, which should be added alongside the self-consistency test, which does not use any ground truth values.